### PR TITLE
pacific: osd/OSDMap: Add health warning if 'require-osd-release' != current release

### PR DESCRIPTION
**Pull request by sseshasa**
backport tracker: https://tracker.ceph.com/issues/53551

---

backport of https://github.com/ceph/ceph/pull/44090
parent tracker: https://tracker.ceph.com/issues/51984

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

File: PendingReleaseNotes [hunk 1]
```diff
@@ -48,6 +48,9 @@
   until we implement a limit on the number of PGs default pools should consume,
   in combination with the 'scale-down' profile.
 
+* A health warning will now be reported if the ``require-osd-release`` flag is not
+  set to the appropriate release after a cluster upgrade.
+
```
Comment (neha-ojha) [inline PendingReleaseNotes:53]: Could you please add a new section for 16.2.8 and add this under it?

Comment (sseshasa) [reply PendingReleaseNotes]: Done.

**Review by neha-ojha — APPROVED** [review]

**Comment by batrick** [comment]
https://github.com/ceph/ceph/pull/44090#issuecomment-995027561

**Comment by sseshasa** [comment]
@vshankar @batrick  Please review the changes to the fs upgrade files. Thanks!

**Comment by vshankar** [comment]
> @vshankar @batrick Please review the changes to the fs upgrade files. Thanks!

Any reason why https://github.com/ceph/ceph/pull/44324 is not a separate commit (with s/quincy/pacific) in this PR?

**Comment by sseshasa** [comment]
> > @vshankar @batrick Please review the changes to the fs upgrade files. Thanks!
> 
> Any reason why #44324 is not a separate commit (with s/quincy/pacific) in this PR?

Sure, I can cherry-pick the commit from #44324 and add one more change to "..volumes/import-legacy/tasks/2-upgrade.yaml". Would that work for you?

**Comment by vshankar** [comment]
> Sure, I can cherry-pick the commit from #44324 and add one more change to "..volumes/import-legacy/tasks/2-upgrade.yaml". Would that work for you?

Yep - that would make it clean.

**Comment by sseshasa** [comment]
@neha-ojha Please review this again as I cherry-picked a subset of commits from #44161.

**Review by batrick — APPROVED** [review]
`fs:upgrade` changes LGTM

**Comment by vshankar** [comment]
> The commit is cherry-picked but I changed the commit message to reflect 'pacific' for correctness. I know that commit messages should not be modified during backports, but in this case I changed it and left a comment in the 'Conflicts' section to be clear about the change. I can revert it if this is not agreeable.

Thanks. Looks good!

**Review by neha-ojha — APPROVED** [review]
nit: It will be good to include an explanation in 7ef6c85d918262f8af60ddee77edfbb28bf2b38e's commit message for why it is not cherry-picked from master.

Needs to go through the entire upgrade suite

**Comment by sseshasa** [comment]
> nit: It will be good to include an explanation in [7ef6c85](https://github.com/ceph/ceph/commit/7ef6c85d918262f8af60ddee77edfbb28bf2b38e)'s commit message for why it is not cherry-picked from master.
> 
> Needs to go through the entire upgrade suite

Updated the commit message and I initiated upgrade tests in teuthology for the ones that were modified.

**Comment by sseshasa** [comment]
### Pacific Upgrade Runs Summary:

All failures seen (2 Nos) are unrelated. 

**cephfs/nofs - PASSED**
1. https://pulpito.ceph.com/sseshasa-2022-01-19_10:12:04-upgrade-wip-53551-pacific-test-distro-default-smithi

**cephfs/featureful_client - PASSED**
1. https://pulpito.ceph.com/sseshasa-2022-01-19_10:10:49-upgrade-wip-53551-pacific-test-distro-default-smithi

**Nautilux-x - PASSED**
1. https://pulpito.ceph.com/sseshasa-2022-01-19_09:03:32-upgrade-wip-53551-pacific-test-distro-default-smithi
   - 17 dead jobs because of infra issue.
2. https://pulpito.ceph.com/sseshasa-2022-01-20_06:56:00-upgrade-wip-53551-pacific-test-distro-default-smithi/
   - All dead jobs from first run passed 

**Octopus-x - 1 Test failed**
1. https://pulpito.ceph.com/sseshasa-2022-01-19_09:05:33-upgrade-wip-53551-pacific-test-distro-default-smithi/
   - 1 job passed.
   - 7/8 dead jobs due to infra issue.
2. https://pulpito.ceph.com/sseshasa-2022-01-20_06:54:06-upgrade-wip-53551-pacific-test-distro-default-smithi/
   - 6/7 dead jobs from first run passed.
   - 1 test failed:
     - https://pulpito.ceph.com/sseshasa-2022-01-20_06:54:06-upgrade-wip-53551-pacific-test-distro-default-smithi/6628741
     - Related to https://tracker.ceph.com/issues/53002. The fix is yet to be backported to pacific.

**cephfs/volumes - FAILED**
1. https://pulpito.ceph.com/sseshasa-2022-01-19_10:13:02-upgrade-wip-53551-pacific-test-distro-default-smithi
   - First attempt failed due to infra issue.
2. https://pulpito.ceph.com/sseshasa-2022-01-20_06:55:03-upgrade-wip-53551-pacific-test-distro-default-smithi
   - Second attempt failed due to https://tracker.ceph.com/issues/53360. Fix is pending in pacific.

**Comment by sseshasa** [comment]
### Teuthology Test Result

http://pulpito.front.sepia.ceph.com/yuriw-2022-01-27_00:01:03-upgrade-wip-yuri2-testing-2022-01-21-0949-pacific-distro-default-smithi/

**Unrelated Failures:**
1. http://pulpito.front.sepia.ceph.com/yuriw-2022-01-27_00:01:03-upgrade-wip-yuri2-testing-2022-01-21-0949-pacific-distro-default-smithi/6642457
- ceph-mon on one node hit the below assertion failure:
  src/mds/FSMap.cc: 868: FAILED ceph_assert(info.compat.writeable(fs->mds_map.compat))
- Tracked by: https://tracker.ceph.com/issues/54064

2. http://pulpito.front.sepia.ceph.com/yuriw-2022-01-27_00:01:03-upgrade-wip-yuri2-testing-2022-01-21-0949-pacific-distro-default-smithi/6642471
 - Tracked by: https://tracker.ceph.com/issues/53360. Fix is pending in pacific. 

The dead jobs are infrastructure related.

**Pull request merged**